app/models/user.rb

```diff
@@ -1590,6 +1590,11 @@ def moderated_with( moderator_action )
       self.spammer = false
       self.suspended_by_user = nil
       unsuspend!
+    elsif moderator_action.action == ModeratorAction::REPLACEUSERNAME
+      new_login = moderator_action.resource_content.to_s.strip
+      return if new_login.blank? || User.exists?(login: new_login)
+      self.login = new_login
+      save!
```

Comment:
This is where you want to use `User.suggest_login`. I guess we could use `resource_content` to do message passing like this, but it's more of a bookkeeping tool.